web/src/main/java/org/fao/geonet/kernel/search/CollectionAvailabilityFilter.java

```diff
@@ -0,0 +1,76 @@
+package org.fao.geonet.kernel.search;
+
+import jeeves.server.context.ServiceContext;
+import jeeves.utils.Log;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.AtomicReaderContext;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.search.*;
+import org.apache.lucene.util.Bits;
+import org.apache.lucene.util.DocIdBitSet;
+import org.fao.geonet.GeonetContext;
+import org.fao.geonet.constants.Geonet;
+
+import java.io.IOException;
+import java.util.BitSet;
+import java.util.Collections;
+import java.util.Set;
+
+public class CollectionAvailabilityFilter extends Filter {
+    private Query _query;
+    private Set<String> _fieldsToLoad;
+    private GeonetContext _geonetContext = null;
+
+    public CollectionAvailabilityFilter(Query query, ServiceContext serviceContext) {
+        this._query = query;
+        _fieldsToLoad = Collections.singleton("_uuid");
+
+        if (serviceContext != null)
```

Comment:
The geonetwork way is... all over! I can get braces over all single lines, but I don't think it was consistent from what I've seen, so I allowed myself to not use braces where I saw fit. It does not conform to our coding standard, I know.